pkg/api/login.go

```diff
@@ -210,7 +210,21 @@ func (hs *HTTPServer) LoginPost(c *models.ReqContext) response.Response {
 	err := bus.Dispatch(c.Req.Context(), authQuery)
 	authModule = authQuery.AuthModule
 	if err != nil {
-		resp = response.Error(401, "Invalid username or password", err)
+		var msg string
+
+		if errors.Is(err, login.ErrTooManyLoginAttempts) {
+			msg = "Too many consecutive incorrect login attempts for user - login for user temporarily blocked"
+		}
+
+		if errors.Is(err, login.ErrInvalidCredentials) {
+			msg = "Invalid username or password"
+		}
+
+		if errors.Is(err, models.ErrUserNotFound) {
+			msg = "User not found"
+		}
+
+		resp = response.Error(401, msg, err)
```

Comment:
If the error is not any of the above cases then the message will be empty (the default string value in Go). So you need [initialise it](https://github.com/grafana/grafana/blob/547310ed1b813393ed0ae6d72789ece084bc4a4c/pkg/api/login.go#L213) to a default like `Invalid username or password`.